File: 451-sort-characters-by-frequency/sort-characters-by-frequency.java
class Solution {
    public String frequencySort(String s) {
    HashMap <Character, Integer> map = new HashMap<>();
    for (char c : s.toCharArray()){
        if(map.containsKey(c)){
            map.put(c, map.get(c) + 1);
        }
        else{
            map.put(c, 1);
        }
    }
    PriorityQueue <Character> pq = new PriorityQueue<> ((a,b) -> map.get(b) - map.get(a));
    pq.addAll(map.keySet());
    StringBuilder ans = new StringBuilder();
    while(!pq.isEmpty()){
        char ch = pq.poll();
        int freq = map.get(ch);
        while(freq-- > 0){
            ans.append(ch);
        
        }
    }
    return ans.toString();
    }
    
}

File: 451-sort-characters-by-frequency/sort-characters-by-frequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        unordered_map <char, int> map;
        for(char ch : s){
            map[ch]++;
        }
        priority_queue <pair <int, char>> pq;
        for(auto& [ch, freq] : map){
            pq.push({freq, ch});
        }

        string ans = "";
        while(!pq.empty()){
            auto [freq, ch] = pq.top();
            pq.pop();
            while(freq > 0){
                ans += ch;
                freq--;
            }
        }
        return ans;
    }
};